### PR TITLE
Add CBMC proof of HandleJobParsingError

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of HandleJobParsingError function

Description
-----------
This PR adds the files required for the CBMC proof of HandleJobParsingError function. 

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is formatted using Uncrustify.
- [ ] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    /* err can only assume values which are of OtaJobParseErr_t enum. */
+    __CPROVER_assume( ( err >= OtaJobParseErrUnknown ) && ( err <= OtaJobParseErrNoActiveJobs ) );
+
+    /* handleJobParsingError can never be called if the parsing err is OtaJobParseErrNone. */
+    __CPROVER_assume( err != OtaJobParseErrNone );
+
+    otaControlInterface.updateJobStatus = updateJobStatusStub;
+
+    ( void ) handleJobParsingError( &fileContext, err );
```
Comment (gshvang) [inline test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c:64]: Aseertin to see if the function executes as expected.

Comment (Bujain) [reply test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c:64]: The return type of the test function viz. handleJobParsingError is void. It does not make sense to have an assertion on void return val.

Comment (gshvang) [reply test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c:64]: Can we check the err value which is being passed as input parameter. If it is getting set in function execution?

Comment (Bujain) [reply test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c:64]: `err` is passed as pass by value and not by pass by reference so the changes to err value would not be reflected in the test harness.

**Review by gshvang — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

File: test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    fileContext.pJobName = &jobNameBuffer;
+
+    /* Havoc otaAgent to non-deterministically set all the bytes in
+     * the structure. */
+    __CPROVER_havoc_object( &otaAgent );
+
+    /* This assumption is used to have nondeterministic length of the jobName String. */
+    __CPROVER_assume( jobNameIdx < OTA_JOB_ID_MAX_SIZE );
+
+    memset( otaAgent.pActiveJobName, 'a', jobNameIdx );
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c:57]: Why is this statement required as long as there is a `\0` at the end?

File: test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+extern OtaAgentContext_t otaAgent;
+extern OtaControlInterface_t otaControlInterface;
+extern void handleJobParsingError( const OtaFileContext_t * pFileContext,
+                                   OtaJobParseErr_t err );
+
+void handleJobParsingError_harness()
+{
+    OtaJobParseErr_t err;
+    OtaFileContext_t fileContext; /* pFileContext can never be NULL as it is passed a field of statically declared otaAgent struct. */
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/handleJobParsingError/handleJobParsingError_harness.c:41]:
```suggestion
    OtaFileContext_t fileContext; /* pFileContext can never be NULL as it is a field of statically declared otaAgent struct. */
```

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**